session/manager.go

```diff
@@ -44,30 +62,58 @@ func (m *Manager) Driver(name ...string) (sessioncontract.Driver, error) {
 	}
 
 	if m.drivers[driverName] == nil {
-		if m.customDrivers[driverName] == nil {
-			return nil, fmt.Errorf("driver [%s] not supported", driverName)
-		}
-
-		m.drivers[driverName] = m.customDrivers[driverName]
+		return nil, fmt.Errorf("driver [%s] not supported", driverName)
 	}
 
 	return m.drivers[driverName], nil
 }
 
-func (m *Manager) Extend(driver string, handler func() sessioncontract.Driver) sessioncontract.Manager {
-	m.customDrivers[driver] = handler()
-	return m
+func (m *Manager) Extend(driver string, handler sessioncontract.Driver) (sessioncontract.Manager, error) {
+	if m.drivers[driver] != nil {
```

Comment:
User can overwrite existing driver.

Reply:
> User can overwrite existing driver.

This seems not suitable with Go's habits, and overwriting the driver may cause GC exceptions.

Reply:
Hmm, makes sense. It's okay to merge custom and built-in drivers.